### PR TITLE
Apply vcvars params for Visual 14 too

### LINKED ISSUE #3045: vcvars_command ignores sdk version for vcversion <= 14
I was doing:

vcvars = tools.vcvars_command(self.settings,winsdk_version="8.1")

but the 8.1 wasn't getting appended

Looking at the code:

```
if int(compiler_version) > 14:
                vcvars_path = os.path.join(vs_path, "VC/Auxiliary/Build/vcvarsall.bat")
                command = ('set "VSCMD_START_DIR=%%CD%%" && '
                           'call "%s" %s' % (vcvars_path, vcvars_arch))
                if winsdk_version:
                    command += " %s" % winsdk_version
                if vcvars_ver:
                    command += " -vcvars_ver=%s" % vcvars_ver
```

version 14 is excluded for some reason. 

however the sdk does seem supported for v14:

```
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC>vcvarsall.bat /?
Error in script usage. The correct usage is:
    vcvarsall.bat [option]
  or
    vcvarsall.bat [option] store
  or
    vcvarsall.bat [option] [version number]
  or
    vcvarsall.bat [option] store [version number]
where [option] is: x86 | amd64 | arm | x86_amd64 | x86_arm | amd64_x86 | amd64_arm
where [version number] is either the full Windows 10 SDK version number or "8.1" to use the windows 8.1 SDK
:
The store parameter sets environment variables to support
  store (rather than desktop) development.
:
For example:
    vcvarsall.bat x86_amd64
    vcvarsall.bat x86_arm store
    vcvarsall.bat x86_amd64 10.0.10240.0
    vcvarsall.bat x86_arm store 10.0.10240.0
    vcvarsall.bat x64 8.1
    vcvarsall.bat x64 store 8.1
:
Please make sure either Visual Studio or C++ Build SKU is installed.
```
The workaround is to manually append the sdk verison

The sdk version is needed when a newer sdk version is installed...projects can fail to build otherwise.

Either way, it might be nice to output an error/warning if someone specifies a setting which is considered not supported so they know why it has no effect.

### PR DESCRIPTION
**Pull request by danimtb**
- [x] Refer to the issue that supports this Pull Request: closes #3045
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. Also adding a description of the changes in the ``changelog.rst`` file. https://github.com/conan-io/docs

### CODE REVIEW

**Comment by danimtb** [comment]
Ready!

File: conans/client/tools/win.py [hunk 1]
```diff
@@ -268,18 +267,18 @@ def vcvars_command(settings, arch=None, compiler_version=None, force=False, vcva
...
                 command = ('set "VSCMD_START_DIR=%%CD%%" && '
                            'call "%s" %s' % (vcvars_path, vcvars_arch))
-                if winsdk_version:
-                    command += " %s" % winsdk_version
-                if vcvars_ver:
-                    command += " -vcvars_ver=%s" % vcvars_ver
             else:
                 vcvars_path = os.path.join(vs_path, "VC/vcvarsall.bat")
                 command = ('call "%s" %s' % (vcvars_path, vcvars_arch))
+        if int(compiler_version) > 12:
```
Comment (lasote) [inline conans/client/tools/win.py]: Confirmed version?

Comment (lasote) [reply conans/client/tools/win.py]: But 2015 is 14. Why 12?

Comment (danimtb) [reply conans/client/tools/win.py]: previous version to 14 was 12. At least this is what `cmake --help` says

Comment (lasote) [reply conans/client/tools/win.py]: ah! ok, > 12. Sorry

Comment (lasote) [reply conans/client/tools/win.py]: Why don't ">=14"?

Comment (danimtb) [reply conans/client/tools/win.py]: yes, you are right!

**Pull request merged**